typing/typecore.ml

```diff
@@ -4625,11 +4632,14 @@ and type_statement ?explanation env sexp =
     exp
   end
 
-and type_unpacks ?in_function env unpacks sbody expected_ty =
+and type_unpacks : ?in_function: Location.t * type_expr ->
+  Env.t -> (string Location.loc * Location.t * Uid.t) list ->
```

Comment:
I'm about to make an annoying comment, so feel free to disregard.
I was thinking that this is again mostly noise IMO, the only parameter whose type I couldn't guess (apart from `?in_function`) is `unpacks`.
And perhaps it would make sense to introduce a record type for this?
```ocaml
type to_unpack = {
  name: string Location.loc;
  loc: Location.t;
  uid: Uid.t
}
```
I don't think this change would be too invasive.

But as I said, feel free to disregard :)

Reply:
Note: part of the reason why I mention this is that I'd find this ugly:
```ocaml
and type_unpacks ?(in_function: Location.t * type_expr) env
  (unpacks : (string Location.loc * Location.t * Uid.t) list) sbody expected_ty =
```
While for some reason this is acceptable:
```ocaml
and type_unpacks ?(in_function: Location.t * type_expr) env (unpacks : to_unpack list) sbody expected_ty =
```

Reply:
I don't think your comments annoying at all! Let me try that idea.